src/models/dml/dml_command_allocator_ring.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Can the dml specific files be under the src directory?
Since there'll be dml code used by the non model code also.